lib/widgets/sign_up_widget.dart

```diff
@@ -0,0 +1,174 @@
+                height: MediaQuery.of(context).size.height - 60,
+                alignment: Alignment.bottomCenter,
+                child: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 25),
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      Form(
+                        key: formKey,
+                        child: Column(
+                          children: [
+                            TextFormField(
+                              validator: (value) {
+                                return value.isEmpty || value.length < 4
+                                    ? AppText.usernameValidatorText
+                                    : null;
+                              },
+                              controller: userNameTextEditingController,
+                              style: Theme.of(context).textTheme.bodyText1,
+                              decoration: InputDecoration(
+                                hintText: AppText.usernameText,
+                              ),
+                            ),
+                            TextFormField(
+                              validator: (value) {
+                                return RegExp(
+                                            r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+")
+                                        .hasMatch(value)
+                                    ? null
+                                    : AppText.emailValidatorText;
+                              },
+                              controller: emailTextEditingController,
+                              style: Theme.of(context).textTheme.bodyText1,
+                              decoration: InputDecoration(
+                                hintText: AppText.emailText,
+                              ),
+                            ),
+                            TextFormField(
+                              obscureText: true,
+                              validator: (value) {
+                                return value.length > 6
+                                    ? null
+                                    : AppText.passwordValidatorText;
+                              },
+                              controller: passwordTextEditingController,
+                              style: Theme.of(context).textTheme.bodyText1,
+                              decoration: InputDecoration(
+                                hintText: AppText.passwordText,
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                      Divider(height: 22),
+                      Divider(height: 22),
+                      GestureDetector(
+                        onTap: () {
+                          signMeUp();
+                        },
+                        child: Container(
```

Comment:
this and every other button on this screen could be also moved to separate build method as in the comment above